src/uct/ib/base/ib_md.c

```diff
@@ -1255,6 +1305,12 @@ static void uct_ib_md_close(uct_md_h uct_md)
     if (md->rcache != NULL) {
         ucs_rcache_destroy(md->rcache);
     }
+    if (md->global_mem.atomic_mr != NULL) {
```

Comment:
need to extract this to common function which would be called from error flow as well